packages/xo-web/src/xo-app/backup/health/index.js

```diff
@@ -66,26 +124,145 @@ const ACTIONS = [
   },
 ]
 
-@addSubscriptions({
-  // used by createGetLoneSnapshots
-  schedules: subscribeSchedules,
-})
-@connectStore({
-  loneSnapshots: createGetLoneSnapshots,
-  legacySnapshots: createGetObjectsOfType('VM-snapshot').filter([
-    (() => {
-      const RE = /^(?:XO_DELTA_EXPORT:|XO_DELTA_BASE_VM_SNAPSHOT_|rollingSnapshot_)/
-      return (
-        { name_label } // eslint-disable-line camelcase
-      ) => RE.test(name_label)
-    })(),
-  ]),
-  vms: createGetObjectsOfType('VM'),
-})
-export default class Health extends Component {
-  render() {
+const Health = decorate([
+  injectIntl,
+  addSubscriptions({
+    // used by createGetLoneSnapshots
+    schedules: cb =>
+      subscribeSchedules(schedules => {
+        cb(keyBy(schedules, 'id'))
+      }),
+    jobs: cb =>
+      subscribeBackupNgJobs(jobs => {
+        cb(keyBy(jobs, 'id'))
+      }),
+  }),
+  connectStore({
+    loneSnapshots: createGetLoneSnapshots,
+    legacySnapshots: createGetObjectsOfType('VM-snapshot').filter([
+      (() => {
+        const RE = /^(?:XO_DELTA_EXPORT:|XO_DELTA_BASE_VM_SNAPSHOT_|rollingSnapshot_)/
+        return (
+          { name_label } // eslint-disable-line camelcase
+        ) => RE.test(name_label)
+      })(),
+    ]),
+    vms: createGetObjectsOfType('VM'),
+  }),
+  provideState({
+    initialState: () => ({
+      backupsByRemote: undefined,
+    }),
+    effects: {
+      initialize({ fetchBackupList }) {
+        return fetchBackupList()
+      },
+      async fetchBackupList() {
+        this.state.backupsByRemote = await listVmBackups(
+          toArray(await getRemotes())
+        )
+      },
+      deleteBackups: ({ fetchBackupList }, backups) => {
+        const nBackups = backups.length
+        return confirm({
+          title: _('deleteBackups', { nBackups }),
+          body: _('deleteBackupsMessage', { nBackups }),
+          icon: 'delete',
+        })
+          .then(() => deleteBackups(backups))
+          .then(fetchBackupList, noop)
+      },
+    },
+    computed: {
+      detachedBackups: ({ backupsByRemote }, { jobs, vms, schedules }) => {
+        if (jobs === undefined || schedules === undefined) {
```

Comment:
What if VMs are not fetched yet?

Reply:
if the VM is not fetched it will be an empty object. So, we can't differentiate if the VM list is not fetched or there is no VMs.